### PR TITLE
Prepare dokka

### PR DESCRIPTION
**Pull request by ZacSweers**
This updates to Dokka 1.4.10 and prepares for a future with Kotlin rewrite and mkdocs

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -99,5 +95,18 @@ subprojects {
...
     }
+
+    if (name != "codegen" && pluginManager.hasPlugin("org.jetbrains.kotlin.jvm")) {
+      apply(plugin = "org.jetbrains.dokka")
+      tasks.named<DokkaTask>("dokkaHtml") {
+        outputDirectory.set(rootDir.resolve("docs/1.x"))
+        dokkaSourceSets.configureEach {
+          skipDeprecated.set(true)
+          externalDocumentationLink {
+            url.set(URL("https://square.github.io/okio/2.x/okio/"))
```
Comment (ZacSweers) [inline build.gradle.kts:106]: Note that 1.x isn't an option because the javadoc is broken https://square.github.io/okio/1.x/okio/

Comment (swankjesse) [reply build.gradle.kts:106]: Yikes. https://github.com/square/okio/issues/775#issuecomment-699266366

**Review by swankjesse — APPROVED** [review]
Presumably we won’t run Dokka on Moshi 1.x?

**Comment by ZacSweers** [comment]
We do already today, just with an older version and only for the moshi-kotlin and code gen artifacts. This HTML option is more useful for a project site

File: build.gradle.kts [hunk 2]
```diff
@@ -24,9 +26,9 @@ buildscript {
 }
 
 plugins {
-  id("com.vanniktech.maven.publish") version "0.12.0" apply false
-  id("org.jetbrains.dokka") version "0.10.1" apply false
-  id("com.diffplug.spotless") version "5.5.0"
+  id("com.vanniktech.maven.publish") version "0.13.0" apply false
+  id("org.jetbrains.dokka") version "1.4.10" apply false
+  id("com.diffplug.spotless") version "5.6.0"
```
Comment (ZacSweers) [inline build.gradle.kts:31]: Opportunistically upped this

**Review by swankjesse — APPROVED** [review]

**Pull request merged**